asteroid/filterbanks/transforms.py

```diff
@@ -371,17 +385,104 @@ def ebased_vad(mag_spec, th_db: int = 40):
     return log_mag > (max_log_mag - th_db)
 
 
-_inputs = {"reim": (take_reim, 1), "mag": (take_mag, 1 / 2), "cat": (take_cat, 1 + 1 / 2)}
-_inputs["real"] = _inputs["reim"]
-_inputs["mod"] = _inputs["mag"]
-_inputs["concat"] = _inputs["cat"]
+def compute_delta(feats: torch.Tensor, dim: int = -1) -> torch.Tensor:
+    """Compute delta coefficients of a tensor.
 
+    Args:
+        feats: Input features to compute deltas with.
+        dim: feature dimension in the feats tensor.
+
+    Returns:
+        Tensor: Tensor of deltas.
 
-_masks = {
-    "reim": (apply_real_mask, 1),
-    "mag": (apply_mag_mask, 1 / 2),
-    "complex": (apply_complex_mask, 1),
-}
-_masks["real"] = _masks["reim"]
-_masks["mod"] = _masks["mag"]
-_masks["comp"] = _masks["complex"]
+    Examples
+        >>> import torch
+        >>> phase = torch.randn(2, 257, 100)
+        >>> # Compute instantaneous frequency
+        >>> inst_freq = compute_delta(phase, dim=-1)
+        >>> # Or group delay
+        >>> group_delay = compute_delta(phase, dim=-2)
+    """
+    if dim != -1:
+        return compute_delta(feats.transpose(-1, dim), dim=-1).transpose(-1, dim)
+    # First frame has nothing. Then each frame is the diff with the previous one.
+    delta = feats.new_zeros(feats.shape)
+    delta[..., 1:] = feats[..., 1:] - feats[..., :-1]
```

Comment:
Wild speculation; could this be more efficient using `F.pad` instead?

Reply:
Maybe. Doesn't seem important to me at this point but you can have a go :wink: